pylabnet/network/client_server/tektronix_tds2004C.py

```diff
@@ -0,0 +1,254 @@
+        val = self._module.extract_params(command, value)
+        return pickle.dumps(val)
+
+
+class Client(ClientBase):
+
+    def reset(self):
+        return self._service.exposed_reset()
+
+    def get_trigger_source(self):
+        return self._service.exposed_get_trigger_source()
+
+    def set_trigger_source(self, trigger_source):
+        return self._service.exposed_set_trigger_source(trigger_source)
+
+    def set_timing_scale(self, scale):
+        return self._service.exposed_set_timing_scale(scale)
+
+    def get_timing_scale(self):
+        return self._service.exposed_get_timing_scale()
+
+    def set_single_run_acq(self):
+        return self._service.exposed_set_single_run_acq()
+
+    def acquire_single_run(self):
+        return self._service.exposed_acquire_single_run()
+
+    def read_out_trace(self, channel, curve_res=1):
+        pickled_trace = self._service.exposed_read_out_trace(
+            channel,
+            curve_res
+        )
+        return pickle.loads(pickled_trace)
+
+    def show_trace(self, channel):
+        return self._service.exposed_show_trace(channel)
+
+    def hide_trace(self, channel):
+        return self._service.exposed_hide_trace(channel)
+
+    def get_channel_attenuation(self, channel):
+        return self._service.exposed_get_channel_attenuation(channel)
+
+    def set_channel_attenuation(self, channel, attenuation):
+        return self._service.exposed_set_channel_attenuation(
+            channel,
+            attenuation
+        )
+
+    def get_channel_scale(self, channel):
+        return self._service.exposed_get_channel_scale(channel)
+
+    def set_channel_scale(self, channel, range):
+        return self._service.exposed_set_channel_scale(channel, range)
+
+    def get_channel_pos(self, channel):
+        return self._service.exposed_get_channel_pos(channel)
+
+    def set_channel_pos(self, channel, pos):
+        return self._service.exposed_set_channel_pos(channel, pos)
+
+    def get_horizontal_position(self):
+        return self._service.exposed_get_horizontal_position()
+
+    def set_horizontal_position(self, hor_pos):
+        return self._service.exposed_set_horizontal_position(hor_pos)
+
+    def trig_level_to_fifty(self):
+        return self._service.exposed_trig_level_to_fifty()
+
+    def get_trigger_level(self):
+        return self._service.exposed_get_trigger_level()
+
+    def set_trigger_level(self, trigger_level):
+        return self._service.exposed_set_trigger_level(trigger_level)
+
+    def query(self, command):
+        query = self._service.exposed_query(command)
+        return pickle.loads(query)
+
+    def write(self, command):
+        return self._service.exposed_write(command)
+
+    def extract_params(self, command, value):
+        val = self._service.exposed_extract_params(command, value)
+        return pickle.loads(val)
+
+    def plot_traces(self, channel_list, curve_res=1, staggered=True, reps=1):
```

Comment:
There should probably be standardized functions for plotting 1D traces (check gui/igui/iplot), probably best to develop these and just wrap them when needed for specific plotting purposes.

Reply:
Yes, good idea. I suggest implementing this in the new branch I'm working on right now (which deals with the waveform comparison), and which is based on this branch. 

I'll try to use `plotly` for this unified plotting function, because I think it will provide more useful analysis tools, like cursors, etc.